### DISCUSSION TITLE
destination set

### DISCUSSION THREAD

**Discussion by ADoublLEN** [eleurent/highway-env#444] [General]
Is there any built-in functions can make the controlled vehicle such as a IDM vehicle to keep following a target vehicle which is not at the same lane with the controlled vehicle? Or like keeping moving for a changing destination, which is defined by the position of the vehicle to follow?

**Comment by eleurent** [comment]
Hi,
No, this feature is not available unfortunately. 

- A steering controller and a speed controller are implemented, so you can use them to track any target defined by a heading angle and a lateral offset. However, this will go straight towards the target vehicle and will not follow the road network, so it is suitable e.g. for a straight road but not for more complicated geometries like roundabout.
- For more complex geometries, there is a route planning mechanism which computes a route in the road network to reach any desired "node" (intersection between roads) and lane index. This can allow you to get e.g. near the closest node of the target vehicle for complex road networks, but it will not handle the tracking in real-time once the two vehicles are close.
- Maybe the two systems can be combined: route planning when the two vehicles are on different road segments, and direct lateral tracking when they are not.

**Comment by eleurent** [comment]
Oh sorry I think I read too fast, I didn't see you were asking for the IDMVehicle specifically.
I guess the IDMVehicle is already doing some kind of tradeoff between reaching a target velocity and not getting below some desired distance of its front vehicle. You can probably just set the front vehicle to any target vehicle and that should take care of the longitudinal following. Likewise for lateral control, IDMVehicles can have a preferred lane which you can set to the target vehicle's lane if they are on the same road segment.
